File: lang/java/src/fun/vac/algorithms/meta/NumberTheory.java
package fun.vac.algorithms.meta;

import fun.vac.algorithms.number_theory.Coprimality;
import fun.vac.algorithms.number_theory.Factorial;
import fun.vac.algorithms.number_theory.FibonacciNumber;
import fun.vac.algorithms.number_theory.GreatestCommonDivisor;
import fun.vac.algorithms.number_theory.LeastCommonMultiple;
import fun.vac.algorithms.number_theory.Parity;
import fun.vac.algorithms.number_theory.Primality;
import fun.vac.algorithms.number_theory.PrimeSieves;

public class NumberTheory {

    private NumberTheory() {}

    public static boolean isCoprime(long m, long n) {
        return Coprimality.reduceToBinaryGCD(m, n);
    }

    public static long factorial(long n) {
        return Factorial.iterativeProcedure(n);
    }

    public static long fibonacci(long n) {
        return FibonacciNumber.iterativeProcedure(n);
    }

    public static long gcd(long m, long n) {
        return GreatestCommonDivisor.iterativeBinaryGCD(m, n);
    }

    public static long lcm(long m, long n) {
        return LeastCommonMultiple.reduceToBinaryGCD(m, n);
    }

    public static boolean isEven(long n) {
        return Parity.bitwiseIsEven(n);
    }

    public static boolean isOdd(long n) {
        return Parity.bitwiseIsOdd(n);
    }

    public static boolean isPrime(long n) {
        return Primality.isPrime(n);
    }

    public static boolean isComposite(long n) {
        return Primality.isComposite(n);
    }

    public static int[] sieveOfPrimes(int n) {
        return PrimeSieves.sieveOfEratosthenes(n);
    }
}

File: lang/nodejs/src/vac.fun/algorithms/meta/NumberTheory.js
"use strict"

const Coprimality           = require("vac.fun/algorithms/number_theory/Coprimality")
const Factorial             = require("vac.fun/algorithms/number_theory/Factorial")
const FibonacciNumber       = require("vac.fun/algorithms/number_theory/FibonacciNumber")
const GreatestCommonDivisor = require("vac.fun/algorithms/number_theory/GreatestCommonDivisor")
const LeastCommonMultiple   = require("vac.fun/algorithms/number_theory/LeastCommonMultiple")
const Parity                = require("vac.fun/algorithms/number_theory/Parity")
const Primality             = require("vac.fun/algorithms/number_theory/Primality")
const PrimeSieves           = require("vac.fun/algorithms/number_theory/PrimeSieves")

exports.isCoprime = (m, n) => {
    return Coprimality.reduceToBinaryGCD(m, n)
}

exports.factorial = (n) => {
    return Factorial.iterativeProcedure(n)
}

exports.fibonacci = (n) => {
    return FibonacciNumber.iterativeProcedure(n)
}

exports.gcd = (m, n) => {
    return GreatestCommonDivisor.iterativeBinaryGCD(m, n)
}

exports.lcm = (m, n) => {
    return LeastCommonMultiple.reduceToBinaryGCD(m, n)
}

exports.isEven = (n) => {
    return Parity.bitwiseIsEven(n)
}

exports.isOdd = (n) => {
    return Parity.bitwiseIsOdd(n)
}

exports.isPrime = (n) => {
    return Primality.isPrime(n)
}

exports.isComposite = (n) => {
    return Primality.isComposite(n)
}

exports.sieveOfPrimes = (n) => {
    return PrimeSieves.sieveOfEratosthenes(n)
}
